template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/raz/RangerRazDatalakeConfigProviderTest.java

```diff
@@ -169,6 +169,37 @@ void getAdditionalServicesWhenRazIsEnabledWithRightCm(String testCaseName, Cloud
         );
     }
 
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("razCloudPlatformAndCmVersionDataProvider")
+    @DisplayName("DL is used with the right CM and Raz is requested and Raz hostgroup is present, Raz service needs to be added to the template")
+    void getAdditionalServicesWhenRazIsEnabledAndRAZHGPresentWithRightCm(String testCaseName, CloudPlatform cloudPlatform, String cmVersion) {
+        ClouderaManagerRepo cmRepo = new ClouderaManagerRepo();
+        cmRepo.setVersion(cmVersion);
+        GeneralClusterConfigs generalClusterConfigs = new GeneralClusterConfigs();
+        generalClusterConfigs.setEnableRangerRaz(true);
+        HostgroupView master = new HostgroupView("master", 0, InstanceGroupType.GATEWAY, List.of());
+        HostgroupView idbroker = new HostgroupView("idbroker", 0, InstanceGroupType.CORE, List.of());
+        HostgroupView razHG = new HostgroupView("razhg", 0, InstanceGroupType.CORE, List.of());
+        TemplatePreparationObject preparationObject = Builder.builder()
+                .withStackType(StackType.DATALAKE)
+                .withCloudPlatform(cloudPlatform)
+                .withProductDetails(cmRepo, List.of())
+                .withGeneralClusterConfigs(generalClusterConfigs)
+                .withHostgroupViews(Set.of(master, idbroker, razHG))
+                .build();
+        Map<String, ApiClusterTemplateService> additionalServices = configProvider.getAdditionalServices(cmTemplateProcessor, preparationObject);
+
+        ApiClusterTemplateService service = additionalServices.get("razhg");
+        List<ApiClusterTemplateRoleConfigGroup> roleConfigGroups = service.getRoleConfigGroups();
+        Assertions.assertAll(
+                () -> assertEquals(2, additionalServices.size()),
+                () -> assertEquals("RANGER_RAZ", service.getServiceType()),
+                () -> assertEquals("ranger-RANGER_RAZ", service.getRefName()),
+                () -> assertEquals("RANGER_RAZ_SERVER", roleConfigGroups.get(0).getRoleType()),
+                () -> assertEquals("ranger-RANGER_RAZ_SERVER", roleConfigGroups.get(0).getRefName())
+        );
```

Comment:
Is the expectation to have RAZ in both `master` and `razhg`? If so, I suggest also checking for `master`. (I know `getAdditionalServicesWhenRazIsEnabledWithRightCm())` kinda covers this.)

Reply:
Yes they will both be present I will check for both in this test.